vpc.tf

```diff
@@ -67,88 +58,42 @@ resource "aws_route_table" "public" {
   }
 }
 
-resource "aws_route_table_association" "public1" {
-  subnet_id      = aws_subnet.public1.id
-  route_table_id = aws_route_table.public.id
-}
-
-resource "aws_route_table_association" "public2" {
-  subnet_id      = aws_subnet.public2.id
+resource "aws_route_table_association" "public" {
+  count = length(local.public_cidr)
+  subnet_id      = aws_subnet.public[count.index].id
   route_table_id = aws_route_table.public.id
 }
 
-resource "aws_eip" "nat1" {
-  vpc      = true
-}
-
-resource "aws_eip" "nat2" {
+resource "aws_eip" "nat" {
+  count = length((local.private_cidr))
   vpc      = true
 }
 
-
-
-resource "aws_nat_gateway" "nat1" {
-  allocation_id = aws_eip.nat1.id
-  subnet_id     = aws_subnet.public1.id
+resource "aws_nat_gateway" "nat" {
+  count = length(local.public_cidr)
+  allocation_id = aws_eip.nat[count.index].id
+  subnet_id     = aws_subnet.public[count.index].id
 
   tags = {
-    Name = "Nat gateway 1"
+    Name = "Nat${count.index}"
   }
 
-  # To ensure proper ordering, it is recommended to add an explicit dependency
-  # on the Internet Gateway for the VPC.
-  depends_on = [aws_internet_gateway.main]
 }
 
-resource "aws_nat_gateway" "nat2" {
-  allocation_id = aws_eip.nat2.id
-  subnet_id     = aws_subnet.public2.id
-
-  tags = {
-    Name = "Nat gateway 2"
-  }
 
-  # To ensure proper ordering, it is recommended to add an explicit dependency
-  # on the Internet Gateway for the VPC.
-  depends_on = [aws_internet_gateway.main]
-}
-
-
-resource "aws_route_table" "private1" {
-  vpc_id = aws_vpc.main.id
-
-  route {
-    cidr_block = "0.0.0.0/0"
-    nat_gateway_id = aws_nat_gateway.nat1.id
-  }
-}
-
-
-
-resource "aws_route_table" "private2" {
+resource "aws_route_table" "private" {
+  count = length(local.private_cidr)
   vpc_id = aws_vpc.main.id
 
   route {
     cidr_block = "0.0.0.0/0"
-    nat_gateway_id = aws_nat_gateway.nat2.id
+    nat_gateway_id = aws_nat_gateway.nat[count.index].id
   }
 }
 
-resource "aws_route_table_association" "private1" {
-  subnet_id      = aws_subnet.private1.id
-  route_table_id = aws_route_table.private1.id
-}
-
-resource "aws_route_table_association" "private2" {
-  subnet_id      = aws_subnet.private2.id
-  route_table_id = aws_route_table.private2.id
-}
-
-
-
-
-
-
-
-
 
+resource "aws_route_table_association" "private" {
+  count = 2
+  subnet_id      = aws_subnet.private[count.index].id
+  route_table_id = aws_route_table.private[count.index].id
+}
```

Comment:
could you please, add a newline at the end of the file and also format your code by running the command `terraform fmt`